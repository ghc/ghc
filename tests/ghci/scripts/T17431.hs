module T17431 (sort) where

sort :: Ord a => [a] -> [a]
sort [] = []
sort (x:xs) = insert x (sort xs)

insert :: Ord a => a -> [a] -> [a]
insert x [] = [x]
insert x (y:ys) | x < y     = x:y:ys
                | otherwise = y:(insert x ys)
