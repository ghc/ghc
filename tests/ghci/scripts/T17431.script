:l T17431
:br 5
sort [3,2,1]
:force x xs _result
