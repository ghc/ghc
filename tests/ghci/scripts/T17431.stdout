Breakpoint 0 activated at T17431.hs:5:15-32
Stopped in T17431.sort, T17431.hs:5:15-32
_result :: [a] = _
x :: a = _
xs :: [a] = [_,_]
*** Ignoring breakpoint [T17431.hs:5:15-32]
*** Ignoring breakpoint [T17431.hs:5:15-32]
*** Ignoring breakpoint [T17431.hs:5:15-32]
x = 3
xs = [2,1]
_result = [1,2,3]
